src/MusicalScore/Graphical/SkyBottomLineCalculator.ts

```diff
@@ -152,6 +164,41 @@ export class SkyBottomLineCalculator {
         this.mBottomLine = this.mBottomLine.map(v => (v - Math.min(...this.mBottomLine)) / unitInPixels + this.StaffLineParent.BottomLineOffset);
     }
 
+    /**
+     * go backwards through the skyline array and find a number so that
+     * we can properly calculate the average
+     * @param start
+     * @param backend
+     * @param color
+     */
+    private findPreviousValidNumber(start: number, tSkyLine: []): number {
+        for (let idx: number = start; idx >= 0; idx--) {
+            if (!isNaN(tSkyLine[idx])) {
+                return tSkyLine[idx];
+            } else {
+                return 0;
+            }
+        }
+    }
```

Comment:
we need to return 0 after the for loop as well, in case no value was not undefined